browser/util/generateReport.ts

```diff
@@ -0,0 +1,105 @@
+/* Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+const months = [
+  undefined,
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'Jun',
+  'Jul',
+  'Aug',
+  'Sep',
+  'Oct',
+  'Nov',
+  'Dec',
+];
+
+// Converting json to csv format
+function convertToText(data) {
+  const report = typeof data !== 'object' ? JSON.parse(data) : data;
+  const month = parseInt(data.date.slice(5, 7), 10);
+  const year = parseInt(data.date.slice(0, 4), 10);
+  let summary =
```

Comment:
I am going to put this here but it goes for the entire file here. The way you are concatenating these strings, while it works, is messy and will make it harder to support in the future. Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals on the multi-line support and clean those newlines out.